bin/build

```diff
@@ -36,13 +44,25 @@ fi
 set -e
 
 pushd $IMAGE_DIR
+  
+  if [[ -n "$GIT_TOKEN" ]] ; then
+    SECRETS_CONTAINER=$(docker run -d --rm busybox /bin/sh -c \
+      "echo $GIT_TOKEN > git_token.txt && httpd -p 8000 -h / && sleep 600")
```

Comment:
Why go through the effort of creating a container to serve up the token if it's going to be visible on the command line for the build script anyway?

Would to be simpler to make it an env var that the script would consume rather than an option?

Reply:
do env vars get baked into the image history?  if so, then we can't use a defined ENV.  What this does is expose a sidecar service during build only that the docker build can call into (and thus store into temporary memory)

Reply:
The purpose of the sidecar container is to hide the token from image history.
Looking at the image history after build, this is what I see for that layer:
```sha256:ab871f0a6bbcecdeb54c7e0a5f27aa22c786a777245e044fcb4f04cec982e83e   12 minutes ago      |9 APPLIANCE_ORG=IBMPrivateCloud APPLIANCE_REF=master GIT_HOST=github.ibm.com GIT_TOKEN_AVAILABLE=true MIQ_ORG=IBMPrivateCloud MIQ_REF=master REPO_PREFIX=bluecf SUI_ORG=IBMPrivateCloud SUI_REF=master /bin/sh -c if [[ -n "$GIT_TOKEN_AVAILABLE" ]] ; then       export GIT_TOKEN=$(wget -O - -q http://localhost:8000/git_token.txt) ;       export GIT_HOST=${GIT_TOKEN}@${GIT_HOST} ;     fi &&     echo $GIT_TOKEN &&     mkdir -p ${APPLIANCE_ROOT} &&     echo "https://${GIT_HOST}/${APPLIANCE_ORG}/${REPO_PREFIX}-appliance/tarball/${APPLIANCE_REF}" &&     curl -L https://${GIT_HOST}/${APPLIANCE_ORG}/${REPO_PREFIX}-appliance/tarball/${APPLIANCE_REF} | tar vxz -C ${APPLIANCE_ROOT} --strip 1 ```

I think that keeps the token hidden